### PR TITLE
Add e2e tests

### PR DESCRIPTION
**Pull request by theofidry**
Subset of #105. Instead of doing the jump of adding e2e tests with the PHAR + the scoping, this PR just takes care of the PHAR e2e tests :)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -33,7 +33,16 @@ script:
   - if [[ $coverage = 1 ]]; then phpdbg -qrr phpunit-6.3.phar --coverage-clover build/logs/clover.xml; else php phpunit-6.3.phar; fi
   - if [[ $lint = 1 ]]; then php php-cs-fixer.phar fix --dry-run --diff --no-ansi; fi
   - if [[ $lint = 1 ]]; then phpstan analyse -l5 --ansi src; fi
-  - if [[ ! $deps && $TRAVIS_PHP_VERSION = "7.1" ]]; then php box.phar build; fi;
+  - |
+    if [[ ! $deps && $TRAVIS_PHP_VERSION = "7.1" ]]; then
+      composer install --no-dev --prefer-dist --no-progress --no-suggest --ansi;
+      composer dump-autoload --classmap-authoritative --no-dev;
```
Comment (dunglas) [inline .travis.yml]: You can merge both lines (`composer install --no-dev --prefer-dist --classmap-authoritative --no-progress --no-suggest --ansi`)

File: .travis.yml [hunk 2]
```diff
@@ -33,7 +33,16 @@ script:
   - if [[ $coverage = 1 ]]; then phpdbg -qrr phpunit-6.3.phar --coverage-clover build/logs/clover.xml; else php phpunit-6.3.phar; fi
   - if [[ $lint = 1 ]]; then php php-cs-fixer.phar fix --dry-run --diff --no-ansi; fi
```
Comment (dunglas) [inline .travis.yml]: the `tests/e2e` directory must be excluded in `.php_cs.dist`.

**Comment by theofidry** [comment]
@dunglas build green PR ready for review

**Comment by dunglas** [comment]
Thank you very much @theofidry!

**Pull request merged**